server/get_payload.go

```diff
@@ -201,52 +240,67 @@ func (m *BoostService) getPayload(log *logrus.Entry, signedBlindedBeaconBlockByt
-			respBytes, err := io.ReadAll(resp.Body)
-			if err != nil {
-				log.WithError(err).Warn("error reading response body")
-				return
-			}
+			var result payloadResult
+			result.success = true
 
-			// Get the response's content type
-			respContentType, _, err := mime.ParseMediaType(resp.Header.Get(HeaderContentType))
-			if err != nil {
-				log.WithError(err).Warn("error parsing response content type")
-				respContentType = MediaTypeJSON
-			}
-			log = log.WithField("respContentType", respContentType)
+			if version == GetPayloadV1 {
+				// Get the resp body content
+				respBytes, err := io.ReadAll(resp.Body)
+				if err != nil {
+					log.WithError(err).Warn("error reading response body")
+					return
+				}
 
-			// Get the response's eth consensus version
-			respEthConsensusVersion := resp.Header.Get(HeaderEthConsensusVersion)
-			log = log.WithField("respEthConsensusVersion", respEthConsensusVersion)
+				// Get the response's content type
+				respContentType, _, err := mime.ParseMediaType(resp.Header.Get(HeaderContentType))
+				if err != nil {
+					log.WithError(err).Warn("error parsing response content type")
+					respContentType = MediaTypeJSON
+				}
+				log = log.WithField("respContentType", respContentType)
 
-			// Decode response
-			response := new(builderApi.VersionedSubmitBlindedBlockResponse)
-			err = decodeSubmitBlindedBlockResponse(respBytes, respContentType, respEthConsensusVersion, response)
-			if err != nil {
-				log.WithError(err).Warn("error decoding bid")
-				return
-			}
+				// Get the response's eth consensus version
+				respEthConsensusVersion := resp.Header.Get(HeaderEthConsensusVersion)
+				log = log.WithField("respEthConsensusVersion", respEthConsensusVersion)
 
-			// Check that the payload matches our request
-			err = verifyPayload(log, request, response)
-			if err != nil {
-				log.WithError(err).Warn("error decoding bid")
-				return
+				// Decode response
+				response := new(builderApi.VersionedSubmitBlindedBlockResponse)
+				err = decodeSubmitBlindedBlockResponse(respBytes, respContentType, respEthConsensusVersion, response)
+				if err != nil {
+					log.WithError(err).Warn("error decoding bid")
+					return
+				}
+
+				// Check that the payload matches our request
+				err = verifyPayload(log, request, response)
+				if err != nil {
+					log.WithError(err).Warn("error verifying payload")
+					return
+				}
+
+				result.response = response
 			}
 
 			// The payload is valid, cancel the request for others
 			requestCtxCancel()
 
-			// We have received a payload, cancel other requests
+			// We have received a valid response, cancel other requests
 			if received.CompareAndSwap(false, true) {
-				resultCh <- response
-				log.Info("received payload from relay")
+				resultCh <- result
+				if version == GetPayloadV1 {
+					log.Info("received payload from relay")
+				} else {
+					log.Info("successfully submitted blinded block to relay")
+				}
```

Comment:
I guess we could just do
`log.Info("successfully submitted blinded block to relay)`